### PR TITLE
Line entity

### PR DESCRIPTION
**Pull request by sethalves**
Add a "Line" entity type.  grab.js uses it to provide a visual indication of who is moving something.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 7a85bfa0d159fcfe6d812f559409f775fae4e70d available at: https://gist.github.com/851fd991c04223773e1f<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4843.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4843.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: f647db3bf0f9cbdfc5eddf08f90037aafd29e22b available at: https://gist.github.com/851fd991c04223773e1f<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4843.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4843.dmg

File: libraries/entities-renderer/src/RenderableLineEntityItem.cpp [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    PerformanceTimer perfTimer("RenderableLineEntityItem::render");
+    assert(getType() == EntityTypes::Line);
+    glm::vec3 position = getPosition();
+    // glm::vec3 center = getCenter();
+    glm::vec3 dimensions = getDimensions();
+    glm::quat rotation = getRotation();
+
+    const float MAX_COLOR = 255.0f;
+
+    glm::vec4 lineColor(getColor()[RED_INDEX] / MAX_COLOR, getColor()[GREEN_INDEX] / MAX_COLOR,
```
Comment (jherico) [inline libraries/entities-renderer/src/RenderableLineEntityItem.cpp]: `toGlm(getColor())` will automatically do this conversion, so...

`glm::vec4 lineColor(toGlm(getColor()), getLocalRenderAlpha())`

**Comment by hifi-gustavo** [comment]
Build console output for commit: 4e679c7e3afcae0f5b89bc6dfab694974babf9b6 available at: https://gist.github.com/851fd991c04223773e1f<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4843.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4843.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 500bb13cbb40b2deb5dbb13665e0fe9073c9c389 available at: https://gist.github.com/851fd991c04223773e1f<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4843.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4843.dmg

**Pull request merged**